### PR TITLE
fix (systemaddon): #3378 update contextmenu on unbookmark/bookmark

### LINKED ISSUE #3378: Highlight context menu bookmark/unbookmark doesn't update context menu or card type
We purposely don't refresh highlights on bookmark add/remove, but it looks like the context menu should be updating. Perhaps the Reducer is failing to update (at least on-bookmark -- screenshot below seems to indicate on-remove-bookmark it updated).

Design has previously said to treat a visited old bookmark as "Visited" but allow removing the bookmark.

It's unclear how the context menu and card type should behave when bookmarking/removing bookmark.

Here's a screenshot of what was a bookmark but just removed it from the context menu:
image

### PR DESCRIPTION
**Pull request by rlr**
@Mardak this mostly fixes it for me. But I am still seeing weird behavior sometimes, or it's just getting late.

Fixes #3378

### CODE REVIEW

File: system-addon/common/Reducers.jsm [hunk 1]
```diff
@@ -253,6 +253,9 @@ function Sections(prevState = INITIAL_STATE.Sections, action) {
             delete newSite.bookmarkGuid;
             delete newSite.bookmarkTitle;
             delete newSite.bookmarkDateCreated;
+            if (newSite.type === "bookmark") {
+              newSite.type = "history";
```
Comment (Mardak) [inline system-addon/common/Reducers.jsm]: We'll need to do the same for adding a bookmark.

**Comment by rlr** [comment]
@Mardak ^^ updated.

I included the `if (!newSite.type || newSite.type === ...` guard because I didn't want trending cards in Top Stories to change, for example. Does that seem reasonable?

**Comment by Mardak** [comment]
Design says reflecting "bookmarked" for pocket stories makes sense. And the behavior is that top stories show "Bookmarked" immediately without moving. On refresh, that card no longer shows up in Top Stories and appears in Highlights (with image and description and Bookmarked). I suppose this can be done as a followup in fixing #3131.

**Comment by as-pine-proxy** [comment]
[Treeherder: a85602b2597643fc17aa8f1983e5ac28780921e0](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=a85602b2597643fc17aa8f1983e5ac28780921e0)

**Pull request merged**